### ISSUE TITLE
Evaluator returns all-NaN metrics + a cascade of "Mean of empty slice" warnings when 0 instances are matched

### ISSUE THREAD

**Issue by anirudh9280** [talmolab/sleap-nn#719]
## Summary
When a model predicts 0 instances (e.g. a bottom-up model that collapsed during training, see #718), `sleap_nn/evaluation.py` computes `NaN` for every metric and emits ~8 numpy `RuntimeWarning: Mean of empty slice` / `invalid value encountered in ...` lines. The NaN-plus-warning spew reads like a metric computation bug rather than "the model produced no detections", which cost us real debugging time.

## Repro
Train a bottom-up model that collapses to zero predictions (concrete config in #718 and talmolab/sleap-roots-training#33), then let `sleap-nn train`'s built-in eval run. On the `val` set with 0 matched instances the eval prints:
```
Matched instances: 0, Unmatched GT: 44
.../sleap_nn/evaluation.py:640: RuntimeWarning: Mean of empty slice
  (also evaluation.py:645, :646, :513, :597, :610, :857, :858 + numpy invalid-value-in-divide)
Average Distance: nan px
dist.p50: nan px  ...  Visibility Recall: nan
```

## Suggested fix
Detect the empty-match case up front (0 matched instances) and report one clear line, e.g. "0 matched instances: no detections, metrics undefined (training likely collapsed)", skipping the metric math and the warning cascade. Optionally write a sentinel into the saved `metrics.*.npz` that downstream tooling can detect as "collapsed" instead of NaN.

## Environment
sleap-nn 0.2.0, Python 3.11.14, PyTorch 2.8.0+cu129.

Context: surfaced alongside #718 while establishing the Tier 1 baseline in talmolab/sleap-roots-training#33. cc @eberrigan

**Comment by anirudh9280** [comment]
Adding a clean reproduction. Running the legacy TF recipe for this dataset in sleap-nn 0.2.0 (bottom-up, full resolution) collapsed on all 3 seeds: the model predicted 0 instances, and evaluation then returned NaN for every metric along with the cascade of RuntimeWarnings (Mean of empty slice / invalid value encountered in divide, e.g. evaluation.py:597,610,640,645,646,857,858).

The practical issue is that a full collapse surfaces as all-NaN metrics rather than an explicit signal, so it first reads as a metrics bug instead of "the model detected nothing." Detecting the empty-match case up front and reporting a single line like "0 matched instances, skipping metric computation" (and skipping the metric math) would make the collapse obvious and remove the warning spam.

The underlying collapse is the bottom-up loss issue (#718); this issue is just about the eval UX when 0 instances match. Repro config and runs are in talmolab/sleap-roots-training#33.

**Comment by gitttt-1234** [comment]
Hi @anirudh9280 !

This is fixed in #720! To use this, install sleap-nn from source:
Option 1: [by cloning the repo](https://nn.sleap.ai/latest/installation/#from-source)
Option 2: with uv: `uv pip install "sleap-nn[torch-cpu] @ git+https://github.com/talmolab/sleap-nn.git@main"`

Thanks,

Divya
